src/main/java/org/datavyu/plugins/nativeosx/AVFoundationMediaPlayer.java

```diff
@@ -0,0 +1,74 @@
+package org.datavyu.plugins.nativeosx;
+
+
+import org.datavyu.plugins.MediaException;
+import org.datavyu.plugins.PlayerStateEvent;
+import org.datavyu.plugins.PlayerStateListener;
+
+import java.awt.*;
+import java.net.URI;
+
+public class AVFoundationMediaPlayer extends NativeOSXMediaPlayer {
+
+  private Container container;
+
+  public AVFoundationMediaPlayer(URI mediaPath, Container container) {
+    super(mediaPath);
+    this.container = container;
+  }
+
+  @Override
+  public void init() {
+    addMediaPlayerStateListener(new PlayerStateListenerImpl());
+    initNative();
+
+    container.addNotify();
+    container.add(nativePlayerCanvas, BorderLayout.CENTER);
+
+    incPlayerCount();
```

Comment:
Why do we need this?

Reply:
This logic was in the implementation of the native osx in Datavyu, it is used to keep count of created players, the player count value will be assigned to the id of the native osx player.

Reply:
Where do we decrement the counter?